src/main/java/gregtech/api/items/metaitem/ElectricStats.java

```diff
@@ -64,14 +66,17 @@ public ActionResult<ItemStack> onItemRightClick(World world, EntityPlayer player
     @Override
     public void onUpdate(ItemStack itemStack, Entity entity) {
         IElectricItem electricItem = itemStack.getCapability(GregtechCapabilities.CAPABILITY_ELECTRIC_ITEM, null);
-        if (!entity.world.isRemote && entity instanceof EntityPlayer && electricItem != null &&
+        if (!entity.world.isRemote && entity instanceof EntityPlayer entityPlayer && electricItem != null &&
                 electricItem.canProvideChargeExternally() &&
                 isInDischargeMode(itemStack) && electricItem.getCharge() > 0L) {
 
-            EntityPlayer entityPlayer = (EntityPlayer) entity;
-            InventoryPlayer inventoryPlayer = entityPlayer.inventory;
+            IInventory inventoryPlayer = entityPlayer.inventory;
             long transferLimit = electricItem.getTransferLimit();
 
+            if (Loader.isModLoaded(GTValues.MODID_BAUBLES)) {
+                inventoryPlayer = BaublesModule.getBaublesWrappedInventory(entityPlayer);
+            }
```

Comment:
Could this mod loaded check be integrated into `BaublesModule#getBaublesWrappedInventory`, or would it cause classloading issues if baubles was not present?

Reply:
It shouldn't cause problems right now, but it could cause problems if something in BaublesModule changes in the future, so I think it is safer to check externally